llvm/include/llvm/DebugInfo/DWARF/DWARFDebugLine.h

```diff
@@ -427,6 +429,9 @@ class DWARFDebugLine {
   using LineTableConstIter = LineTableMapTy::const_iterator;
 
   LineTableMapTy LineTableMap;
+
+public:
+  inline static bool ReportApproximateLine = false;
```

Comment:
mutable `static` variables are concerning - that makes the code not thread safe/re-entrant, etc.

Please don't do this - and pass down the request through some parameter/non-static member state. (we have other parameters for line queries, perhaps this can be bundled along with those, or as an optional parameter)

Reply:
Yes, I have created a structure by the name ```Approximate``` to cover this as member of the said struct  and passing object of struct as parameter to functions.